### PR TITLE
Initial clean-up of builtins

### LINKED ISSUE #119: zip() now returns an iterator
https://docs.python.org/3.0/whatsnew/3.0.html#views-and-iterators-instead-of-lists

### LINKED ISSUE #120: map() and filter() return iterators
https://docs.python.org/3.0/whatsnew/3.0.html#views-and-iterators-instead-of-lists

### LINKED ISSUE #121: The builtin basestring abstract type was removed
https://docs.python.org/3.0/whatsnew/3.0.html#text-vs-data-instead-of-unicode-vs-8-bit

### LINKED ISSUE #122: Cleanup of the sys module: removed sys.exitfunc(), sys.exc_clear(), sys.exc_type, sys.exc_value, sys.exc_traceback
https://docs.python.org/3.0/whatsnew/3.0.html#library-changes

### LINKED ISSUE #123: Moved intern() to sys.intern()
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### LINKED ISSUE #124: Removed: apply(). Instead of apply(f, args) use f(*args).
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### LINKED ISSUE #125: Removed coerce(). This function no longer serves a purpose now that classic classes are gone.
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### LINKED ISSUE #126: Removed execfile(). Instead of execfile(fn) use exec(open(fn).read()).
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### LINKED ISSUE #127: Removed file. Use open().
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### LINKED ISSUE #128: Removed reduce(). Use functools.reduce()
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### LINKED ISSUE #129: Removed reload(). Use imp.reload().
https://docs.python.org/3.0/whatsnew/3.0.html#builtins

### PR DESCRIPTION
**Pull request by slide**
Fix #119, Fix #120, Fix #121, Fix #122, Fix #123, Fix #124, Fix #125, Fix #126, Fix #127, Fix #128, Fix #129

### CODE REVIEW

File: Src/IronPython/Modules/Builtin.cs [hunk 1]
```diff
@@ -942,30 +761,12 @@ public static partial class Builtin {
...
-            return eval(context, raw_input(context, prompt));
-        }
-
         public static PythonType @int {
             get {
                 return DynamicHelpers.GetPythonTypeFromType(typeof(int));
             }
         }
 
-        public static string intern(object o) {
```
Comment (slozier) [inline Src/IronPython/Modules/Builtin.cs]: Did you move this to the sys module? I'm not seeing it.

File: Src/IronPython/Runtime/Operations/PythonOps.cs [hunk 2]
```diff
@@ -541,12 +541,25 @@ public static partial class PythonOps {
...
+            return NotImplementedType.Value;
+        }
+
         internal static int GetSequenceMultiplier(object sequence, object count) {
             int icount;
             if (!Converter.TryConvertToIndex(count, out icount)) {
                 PythonTuple pt = null;
                 if (count is OldInstance || !DynamicHelpers.GetPythonType(count).IsSystemType) {
-                    pt = Builtin.TryCoerce(DefaultContext.Default, count, sequence) as PythonTuple;
+                    pt = TryCoerce(DefaultContext.Default, count, sequence) as PythonTuple;
```
Comment (slozier) [inline Src/IronPython/Runtime/Operations/PythonOps.cs]: I think `__coerce__` is gone. Maybe we can get rid of this?

File: Src/IronPython/Modules/Builtin.cs (hunk 1 above)
Comment (slide) [inline Src/IronPython/Modules/Builtin.cs]: You're right, I missed the add to sys.cs, I'll push an update.

File: Src/IronPython/Runtime/Operations/PythonOps.cs (hunk 2 above)
Comment (slide) [inline Src/IronPython/Runtime/Operations/PythonOps.cs]: Yes, it is gone, but I was going to remove it as part of the removal of OldClass and friends.

**Pull request merged**